e2e/test/iothub/service/DirectMethodE2ETests.cs

```diff
@@ -0,0 +1,136 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Net.Http;
+using System.Threading;
+using System.Threading.Tasks;
+using FluentAssertions;
+using FluentAssertions.Specialized;
+using Microsoft.Azure.Devices.E2ETests.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using Microsoft.Azure.Devices.Client;
+using System.Reflection;
+using Microsoft.Identity.Client;
+
+namespace Microsoft.Azure.Devices.E2ETests.iothub.service
+{
+    [TestClass]
+    [TestCategory("E2E")]
+    [TestCategory("IoTHub")]
+    public class DirectMethodE2ETests : E2EMsTestBase
+    {
+        private readonly string _devicePrefix = $"{nameof(DirectMethodE2ETests)}_";
+        private readonly string _modulePrefix = $"{nameof(DirectMethodE2ETests)}_Module_";
+
+        [TestMethod]
+        [Timeout(TestTimeoutMilliseconds)]
+        public async Task DirectMethodsClient_InvokeAsync_DeviceDoesNotExist()
+        {
+            // arrange
+            IotHubServiceClient serviceClient = TestDevice.ServiceClient;
+
+            var methodInvocation = new DirectMethodServiceRequest("someDirectMethod");
+
+            // act
+            Func<Task> act = async () => await serviceClient.DirectMethods.InvokeAsync("someNonexistentDevice", methodInvocation);
+
+            // assert
+            ExceptionAssertions<IotHubServiceException> errorContext = await act.Should().ThrowAsync<IotHubServiceException>();
+            errorContext.And.StatusCode.Should().Be(System.Net.HttpStatusCode.NotFound);
+            errorContext.And.IsTransient.Should().BeFalse();
```

Comment:
nit: one extra line between comment and code block
```suggestion
            // assert
            
            ExceptionAssertions<IotHubServiceException> errorContext = await act.Should().ThrowAsync<IotHubServiceException>();
            errorContext.And.StatusCode.Should().Be(System.Net.HttpStatusCode.NotFound);
            errorContext.And.IsTransient.Should().BeFalse();
```

Reply:
Do we need to check the `ErrorCode` [property ](https://github.com/Azure/azure-iot-sdk-csharp/blob/previews/v2/iothub/service/src/Exceptions/IotHubServiceException.cs#L91)here as well?